### ISSUE TITLE
Hanging on aggregate reports from google.com

### ISSUE THREAD

**Issue by fraserc182** [domainaware/parsedmarc#110]
This may or not be an issue with parsedmarc but  I have encountered an issue where google.com reports are hanging when parsedmarc is trying to parse them.
Eventually I have to interrupt the service and this is the traceback: 

`^CTraceback (most recent call last):
  File "/usr/bin/parsedmarc", line 11, in <module>
    load_entry_point('parsedmarc==6.3.6', 'console_scripts', 'parsedmarc')()
  File "/usr/lib/python3.4/site-packages/parsedmarc/cli.py", line 539, in _main
    strip_attachment_payloads=sa
  File "/usr/lib/python3.4/site-packages/parsedmarc/__init__.py", line 1032, in get_dmarc_reports_from_inbox
    ["RFC822"])[msg_uid]
  File "/usr/lib64/python3.4/site-packages/imapclient/imapclient.py", line 1225, in fetch
    typ, data = self._imap._command_complete('FETCH', tag)
  File "/usr/lib64/python3.4/imaplib.py", line 957, in _command_complete
    typ, data = self._get_tagged_response(tag)
  File "/usr/lib64/python3.4/imaplib.py", line 1077, in _get_tagged_response
    self._get_response()
  File "/usr/lib64/python3.4/imaplib.py", line 1032, in _get_response
    data = self.read(size)
  File "/usr/lib64/python3.4/imaplib.py", line 264, in read
    return self.file.read(size)
  File "/usr/lib64/python3.4/socket.py", line 378, in readinto
    return self._sock.recv_into(b)
KeyboardInterrupt`

This only seems to happen for specific domains and I am not sure why. Is there any other logging I can turn on to see what is happening and why this is hanging for so long?

I have already ran the command with --debug and it doesn't show any useful output. Just it gets to message x of x and then just hangs there.
Happy to share the dmarc report if it will help.

**Comment by seanthegeek** [comment]
Others with the same problem report success reducing the DNS timeout setting to a very low value, like 0.5. Let me know if that works for you, and I'll change the default.

**Comment by fraserc182** [comment]
Thanks for getting back to me so fast.
I added dns_timeout = 0.5 to the conf file but it still is hanging on the reports.

**Comment by seanthegeek** [comment]
Attach the problematic report here by dragging and dropping the file into the comment field, and I'll take a look.

**Comment by fraserc182** [comment]
Before I waste your time crawling through reports, I realised you have fixed an issue similar to this in 6.5.0: https://github.com/domainaware/parsedmarc/issues/79

I am on version 6.3.6 so I will update first of all and let you know what the outcome is.
I will upload the file regardless.
[google.com!postcodeloterij.nl!1568764800!1568851199.zip](https://github.com/domainaware/parsedmarc/files/3635478/google.com.postcodeloterij.nl.1568764800.1568851199.zip)

**Comment by fraserc182** [comment]
Why was this closed? Is there anything else I can do here?

**Comment by seanthegeek** [comment]
Ah, sorry. I just saw your edit. I'll look into this tonight

**Comment by seanthegeek** [comment]
This is strange. I just ran the same the report you attached, and parsedmarc returned output in 1 minute, 18 seconds using the default settings. Maybe something on your network or OS is slowing down your DNS queries?

What OS are you running?

Here's the output:
[google.com!postcodeloterij.nl!1568764800!1568851199.json.zip](https://github.com/domainaware/parsedmarc/files/3639063/google.com.postcodeloterij.nl.1568764800.1568851199.json.zip)

**Comment by fraserc182** [comment]
It's CentOS 7 we are running and it does seem that this is in relation to a few new domains we added to our dmarc record.
I think it's safe to say this isn't a parsedmarc issue in this case. I'm happy to close this and when I eventually find out the problem I'll update this ticket.

Thanks a lot of looking into it for me, very much appreciate the work you've put into this!

**Comment by fraserc182** [comment]
I think I have finally discovered the issue I am running into here.

When parsing these reports and only these reports, parsedmarc appears to be reaching out to 1.1.1.1 instead of the nameservers I have specified in the config file.

In our firewall, each time I try to parse this specific report I can see requests on port 53 being blocked by our firewall.

Why is parsedmarc not trying the nameservers I have specified?
